### ISSUE TITLE
when to release detection and segmentation models?

### ISSUE THREAD

**Issue by xiaohu2015** [facebookresearch/mae#2]

**Comment by BIGBALLON** [comment]
from the paper:

```
Looking beyond ViT, we hope our practices and observations will serve as 
a blueprint for future work comparing
pre-training methods for more advanced ViT derivatives,
like Swin [29] and MViT [12]. To facilitate community development we will release code in Detectron2
```
It will be released in Detectron2. But no one knows when it will be released. see https://github.com/facebookresearch/detectron2/issues/3866

**Comment by DerrickWang005** [comment]
Followed by the segmentation code of the ibot, I obtain 42.3 mIoU on ADE20K, which is extremely lower than the result in the paper. So, when to release detection and segmentation models? I'll be appreciated that.

**Comment by implus** [comment]
We release the codes for semantic segmentation using MAE which reproduces the results reported in paper (48.15 for ViT-Base). See details in [mae_segmentation](https://github.com/implus/mae_segmentation).

**Comment by Pter61** [comment]
So, when to release the detection code?

**Comment by BIGBALLON** [comment]
Hi, guys, is there any progress?

**Comment by junchen14** [comment]
hope to see this soon!!!!

guys, do you have the code that can reproduce the results from paper?

**Comment by rbgirshick** [comment]
We delayed our code release plans due to the ECCV deadline. After ECCV, we will prioritize releasing code. Sorry for the wait.

**Comment by Pter61** [comment]
> We delayed our code release plans due to the ECCV deadline. After ECCV, we will prioritize releasing code. Sorry for the wait.

Thank you for your kindly reply and fantastic work!

**Comment by CauchyFood** [comment]
> We delayed our code release plans due to the ECCV deadline. After ECCV, we will prioritize releasing code. Sorry for the wait.

When can you release this code?

**Comment by Yuxin-CV** [comment]
Hello guys! @xiaohu2015, @BIGBALLON, @DerrickWang005, @Pter61, @junchen14

We successfully reproduce the [object detection & instance segmentation results](https://github.com/hustvl/MIMDet#mask-r-cnn) (50.4 box AP & 44.9 mask AP for ViT-Base) reported in [Benchmarking Detection Transfer Learning with Vision Transformers](https://arxiv.org/abs/2111.11429) using the MAE representations. Code & models are open-sourced.

We also propose MIMDet with better performance (_**51.7 box AP & 46.2 mask AP**_ for ViT-Base, _**54.3 box AP & 48.2 mask AP**_ for ViT-Large) and faster converge speed (36 epochs of fine-tuning). See more details in MIMDet ([paper](https://arxiv.org/abs/2204.02964), [code & models](https://github.com/hustvl/MIMDet)).

**Comment by rbgirshick** [comment]
We've [released code](https://github.com/facebookresearch/detectron2/tree/main/projects/ViTDet) for [Exploring Plain Vision Transformer Backbones for Object Detection](https://arxiv.org/abs/2203.16527), which is an extension of the detection models we benchmarked for the MAE paper.
